tools/topology/topology1/m4/dai.m4

```diff
@@ -155,7 +155,7 @@ define(`DO_DAI_CONFIG',
 `'
 `	id		"'$3`"'
 `'
-`	ifelse($1, `SSP', $5, $1, `HDA', $5, $1, `ALH', $5, $1, `ESAI', $5, $1, `SAI', $5, $1, `AFE', $5, `}')'
+`	ifelse($1, `SSP', $5, $1, `HDA', $5, $1, `ALH', $5, $1, `ESAI', $5, $1, `SAI', $5, $1, `AFE', $5, $1, `ACP', $5, $1, `ACPSP', $5, $1, `ACPDMIC', $5, `}')'
```

Comment:
that line is starting to become unmaintainable. @ranj063 should we start cleaning this up?

Reply:
This should really be one of the last changes to M4 prior to deprecation.

Reply:
wow. this check really needs to be cleaned up. The interesting part is we've covered almost all BE DAI types and all of them add the $5 argument even without bothering to check if there's even a 5th argument supplied. I will submit a PR

Reply:
@ranj063 do you have a PR number ?

Reply:
not yet @lgirdwood. will submit one tomorrow